src/nupic/math/ArrayAlgo.hpp

```diff
@@ -5078,7 +4979,7 @@ namespace nupic {
       // skip the asm. 
       if (n1 > 0) { 
 
-  #if defined (NTA_PLATFORM_darwin86) || defined(NTA_PLATFORM_linux32)
```

Comment:
> I don't know..if it won't, we should add (not win32)

Yeah, please add this condition to similar pieces of code. Otherwise, this will cause future problems to Windows build.. :-/